src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -486,21 +486,30 @@ let create_inotifylib ~scheduler =
   Inotify_lib.add inotify (Lazy.force Fs_sync.special_dir);
   { kind = Inotify inotify; sync_table }
 
-let fsevents_callback (scheduler : Scheduler.t) ~f events =
+let fsevents_callback ?exclusion_paths (scheduler : Scheduler.t) ~f events =
+  let skip_path =
+    (* excluding a [path] will exclude children under [path] but not [path]
+       itself. Hence we need to skip [path] manually *)
+    match exclusion_paths with
+    | None -> fun _ -> false
+    | Some paths -> fun p -> List.mem paths p ~equal:Path.equal
```

Comment:
Out of curiosity, don't you have issues with case insensitivity on macOS when comparing paths? (I'm asking because this is an issue in #6087...)

Reply:
There's probably issues, but this is nowhere near the only place with issues then. Personally, I don't think it's worth handling case insensitivity systematically.